apis/kms/v1alpha1/cryptokey_types.go

```diff
@@ -0,0 +1,465 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// CryptoKeyParameters defines parameters for a desired KMS CryptoKey
+// https://cloud.google.com/kms/docs/reference/rest/v1/projects.locations.keyRings.cryptoKeys
+type CryptoKeyParameters struct {
+	// KeyRing: The RRN of the KeyRing to which this CryptoKey belongs,
+	// provided by the client when initially creating the CryptoKey.
+	// +optional
+	// +immutable
+	KeyRing *string `json:"keyRing,omitempty"`
+
+	// KeyRingRef references a KeyRing and retrieves its URI
+	// +optional
+	// +immutable
+	KeyRingRef *xpv1.Reference `json:"keyRingRef,omitempty"`
+
+	// KeyRingSelector selects a reference to a KeyRing
+	// +optional
+	// +immutable
+	KeyRingSelector *xpv1.Selector `json:"keyRingSelector,omitempty"`
+
+	// Labels: Labels with user-defined metadata. For more information,
+	// see
+	// [Labeling Keys](/kms/docs/labeling-keys).
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// Purpose: Immutable. The immutable purpose of this CryptoKey.
+	//
+	// Possible values:
+	//   "CRYPTO_KEY_PURPOSE_UNSPECIFIED" - Not specified.
+	//   "ENCRYPT_DECRYPT" - CryptoKeys with this purpose may be used
+	// with
+	// Encrypt and
+	// Decrypt.
+	//   "ASYMMETRIC_SIGN" - CryptoKeys with this purpose may be used
+	// with
+	// AsymmetricSign and
+	// GetPublicKey.
+	//   "ASYMMETRIC_DECRYPT" - CryptoKeys with this purpose may be used
+	// with
+	// AsymmetricDecrypt and
+	// GetPublicKey.
+	// +immutable
+	Purpose string `json:"purpose"`
```

Comment:
Could consider a kubebuilder enum here :)

Reply:
Good point, I didn't know that :)